driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/TransactionFailureTest.java

```diff
@@ -19,32 +19,34 @@
 import com.mongodb.ClientSessionOptions;
 import com.mongodb.MongoClientException;
 import org.bson.Document;
-import org.junit.Before;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
 import reactor.core.publisher.Mono;
 
 import static com.mongodb.ClusterFixture.TIMEOUT_DURATION;
 import static com.mongodb.ClusterFixture.isSharded;
 import static com.mongodb.ClusterFixture.serverVersionLessThan;
-import static org.junit.Assume.assumeTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
 
 public class TransactionFailureTest extends DatabaseTestCase {
     public TransactionFailureTest() {
     }
 
-    @Before
+    @BeforeEach
     public void setUp() {
         assumeTrue(canRunTests());
         super.setUp();
     }
 
-    @Test(expected = MongoClientException.class)
+    @Test
     public void testTransactionFails() {
-
-        try (ClientSession clientSession = createSession()) {
-            clientSession.startTransaction();
-            Mono.from(collection.insertOne(clientSession, Document.parse("{_id: 1, a: 1}"))).block(TIMEOUT_DURATION);
-        }
+        assertThrows(MongoClientException.class, () -> {
+            try (ClientSession clientSession = createSession()) {
+                clientSession.startTransaction();
+                Mono.from(collection.insertOne(clientSession, Document.parse("{_id: 1, a: 1}"))).block(TIMEOUT_DURATION);
```

Comment:
Better to narrow the scope of assertThrows to just the statement that is expected to throw (I've found test bugs by doing that).